### PR TITLE
Read environment vars from .env

### PR DESCRIPTION
**Pull request by vkurup**
Uses django-dotenv to load environment variables from `.env`. It does
this from manage.py (for local runserver use and management commands)
and from wsgi.py (for gunicorn eventually).

Updates the README to remove need for venv postactivate hook.

Adds .env to gitignore.

Addresses #166

### CODE REVIEW

File: project_name/wsgi.py [hunk 1]
```diff
@@ -11,6 +11,9 @@
 
 from django.core.wsgi import get_wsgi_application
 
+import dotenv
+dotenv.read_dotenv(os.path.join(os.path.dirname(os.path.dirname(__file__)), '.env'))
```
Comment (mlavin) [inline project_name/wsgi.py]: I think `os.path.abspath(os.path.join(os.path.dirname(__file__), os.pardir, '.env'))` is more readable than nested `dirname` calls.

File: README.rst [hunk 2]
```diff
@@ -38,16 +38,13 @@ necessary requirements::
...
 
     ({{ project_name }})$ cp {{ project_name }}/settings/local.example.py {{ project_name }}/settings/local.py
-    ({{ project_name }})$ echo "export DJANGO_SETTINGS_MODULE={{ project_name }}.settings.local" >> $VIRTUAL_ENV/bin/postactivate
-    ({{ project_name }})$ echo "unset DJANGO_SETTINGS_MODULE" >> $VIRTUAL_ENV/bin/postdeactivate
-
-Exit the virtualenv and reactivate it to activate the settings just changed::
-
-    ({{ project_name }})$ deactivate
-    $ workon {{ project_name }}
+    ({{ project_name }})$ echo "DJANGO_SETTINGS_MODULE={{ project_name }}.settings.local" > .env
```
Comment (dpoirier) [inline README.rst]: This is so much better than having to hack the virtualenv's postactivate script.

**Comment by dpoirier** [comment]
I'm not a big fan of joining os.pardir rather than explicitly writing os.path.dirname() to get the directory name.  But I might break this into multiple steps to make it clearer, maybe add an import to shorten things:

    from os.path import dirname, join

    project_dir = dirname(dirname(__file__))
    dotenv.read_dotenv(join(project_dir), '.env')

Or even

    dotenv.read_dotenv(dirname(__file__) + '/../.env')

It's not like we're going to run this anywhere that pardir is going to be different.

**Comment by vkurup** [comment]
I have to admit that whenever I'm doing stuff with `os.join` or `dirname`, I have to look up the docs or try it in the shell to make sure I know how it works, so all of these version look equally easy/difficult to understand (for me).

@mlavin do either of Dan's versions look ok to you?

**Comment by mlavin** [comment]
I didn't mean to turn this into a giant debate about path creation. Either are fine to me.

**Comment by vkurup** [comment]
Updated, if someone wants to take one last peek :)

**Comment by mlavin** [comment]
:space_invader: :space_invader:

**Pull request merged**